ld-pruning-wf.wdl

```diff
@@ -0,0 +1,440 @@
+
+		Rscript /usr/local/analysis_pipeline/R/merge_gds.R merge_gds.config
+	>>>
+	# Estimate disk size required
+	Int gds_size = ceil(size(gdss, "GB"))
+	Int finalDiskSize = gds_size * 3 + addldisk
+	String filename = select_first([out_prefix, "merged"])
+
+	runtime {
+		cpu: cpu
+		disks: "local-disk " + finalDiskSize + " HDD"
+		docker: "uwgac/topmed-master:2.10.0"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	
+	output {
+		File merged_gds_output = filename + ".gds"
+		File config_file = "merge_gds.config"
+	}
+}
+
+# [4] check_merged_gds -- check a merged GDS file against its inputs
+#
+# This code is nearly identical to the implementation of check_gds in vcf-to-gds-wf.wdl
+# because, like check_gds, you have to remove the chr number of your input file and
+# instead pass it on the command line. It begs the question of "why not just pass in
+# the full name of the file since it's only looking for one file?" but alas it must
+# be passed in this way due to how the R script works.
+#
+task check_merged_gds {
+	input {
+		File gds
+		File merged
+
+		#runtime attributes
+		Int addldisk = 1
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	command <<<
+	set -eux -o pipefail
+
+	python << CODE
+	import os
+	gds = "~{gds}"
+	gds_first_part = gds.split('chr')[0]
+	gds_second_part = gds.split('chr')[1]
+
+	# grab the chr number
+	g = open("chr_number", "a")
+	if(unicode(str(gds_second_part[1])).isnumeric()):
+		# two digit number
+		g.write(gds_second_part[:2])
+		gds_second_part = gds_second_part[2:]
+	else:
+		# one digit number or Y/X/M
+		g.write(gds_second_part[0])
+		gds_second_part = gds_second_part[1:]
+	g.close()
+
+	gds_name = gds_first_part + 'chr ' + gds_second_part
+	f = open("check_merged_gds.config", "a")
+	f.write('gds_file "' + gds_name + '"\n')
+	f.write('merged_gds_file "~{merged}"\n')
+	f.close
+	exit()
+	CODE
+
+	echo "Setting chromosome number"
+	BASH_CHR=$(<chr_number)
+	echo "Chromosme number is ${BASH_CHR}"
+
+	R -q --vanilla < /usr/local/analysis_pipeline/R/check_merged_gds.R --args check_merged_gds.config --chromosome ${BASH_CHR}
+	>>>
+	# Estimate disk size required
+	Int gds_size = ceil(size(gds, "GB"))
+	Int finalDiskSize = gds_size * 3 + addldisk
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master:2.10.0"
+		disks: "local-disk " + finalDiskSize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File config_file = "check_merged_gds.config"
+	}
+
+}
+
+workflow b_ldpruning {
+	input {
+		Array[File] gds_files
+		String? out_prefix
+		File? sample_include_file
+		File? variant_include_file
+	}
+
+	scatter(gds_file in gds_files) {
+		call ld_pruning {
+			input:
+				gds = gds_file,
+				sample_include_file = sample_include_file,
+				variant_include_file = variant_include_file,
+				out_prefix = out_prefix
+		}
+	}
+
+	# CWL uses a dotproduct scatter; this is the closest WDL equivalent
+	scatter(gds_n_varinc in zip(gds_files, ld_pruning.ld_pruning_output)) {
```

Comment:
Shouldn't this call be to
```
scatter(gds_n_varinc in zip(ld_pruning.ld_pruning_output, variant_include_file)) {
```
It seems that you are including the original GDS and the LD Pruned GDS not the LD pruned GDS and the variants to include.

Reply:
And if that is the case, then you don't actually need to zip() this.

Reply:
LD Pruning does not actually output GDS files. LD Pruning outputs RData files saying "here are the places to cut out." It is the subset GDS task who then takes in the GDS files and the LD Pruning RData files and does the actual cutting.